lib/test_after_commit/with_transaction_state.rb

```diff
@@ -0,0 +1,12 @@
+# disable parts of the sync code that starts looping
+module TestAfterCommit
+  module WithTransactionState
+    def sync_with_transaction_state
+      @reflects_state ||= []
+      @reflects_state[0] = true
+      super
+    end
+
+    ActiveRecord::Base.prepend(self)
```

Comment:
You're prepending the module after it's required in `test_after_commit.rb`, it doesn't need to be prepended here too.